dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -729,6 +730,42 @@ def test_correct_total_sample_size_and_counts_and_mutability(self):
         self.assertEqual(col_one_len, len(data['NAME']))
         self.assertEqual(col_two_len, len(data['VALUE']))
 
+    def test_save_and_load(self):
+        datapth = "dataprofiler/tests/data/"
+        test_files = ["csv/guns.csv", "csv/iris.csv"]
+
+        def _clean_report(report):
+            data_stats = report["data_stats"]
+            for key in data_stats:
+                stats = data_stats[key]["statistics"]
+                if "histogram" in stats:
+                    if "bin_counts" in stats["histogram"]:
+                        stats["histogram"]["bin_counts"] = \
+                            stats["histogram"]["bin_counts"].tolist() 
+                    if "bin_edges" in stats["histogram"]:
+                        stats["histogram"]["bin_edges"] = \
+                            stats["histogram"]["bin_edges"].tolist() 
+            return report
+
+        for test_file in test_files:
+            # Create Data and Profiler objects
+            data = dp.Data(os.path.join(datapth, test_file))
+            save_profile = dp.Profiler(data)
+            
+            # Save and Load profile with Mock IO
+            save_profile.save()
+            load_profile = dp.Profiler.load()
+            #with mock.patch('builtins.open', mock.mock_open()) as m: 
+            #    save_profile.save()
+            #with mock.patch('builtins.open', mock.mock_open(
+            #    read_data = m().write.call_args[0][0])) as m:
+            #    load_profile = dp.Profiler.load()
```

Comment:
These mocks should go around the function at the top so they can apply throughout the calls.
Likely you will have to mock the DataLabeler as well as it uses the `open` command too. 

We should avoid creating objects during the tests

Reply:
Fixed in new commit